### ISSUE TITLE
'ascii' codec can't decode byte 0xe8 in position 1314: ordinal not in range(128)

### ISSUE THREAD

**Issue by JZLeung** [certbot/certbot#5757]
when i try to run the command `certbot --nginx`, it didn't work:
```bash
root@localhost:~# certbot --nginx
Saving debug log to /var/log/letsencrypt/letsencrypt.log
An unexpected error occurred:
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe8 in position 1314: ordinal not in range(128)
Please see the logfiles in /var/log/letsencrypt for more details.
```

the log's detail are these: 
```
root@localhost:~# cat  /var/log/letsencrypt/letsencrypt.log
2018-02-22 14:27:50,994:DEBUG:certbot.main:certbot version: 0.21.1
2018-02-22 14:27:50,994:DEBUG:certbot.main:Arguments: []
2018-02-22 14:27:50,994:DEBUG:certbot.main:Discovered plugins: PluginsRegistry(PluginEntryPoint#apache,PluginEntryPoint#manual,PluginEntryPoint#nginx,PluginEntryPoint#null,PluginEntryPoint#standalone,PluginEntryPoint#webroot)
2018-02-22 14:27:51,010:DEBUG:certbot.log:Root logging level set at 20
2018-02-22 14:27:51,011:INFO:certbot.log:Saving debug log to /var/log/letsencrypt/letsencrypt.log
2018-02-22 14:27:51,036:DEBUG:certbot.plugins.selection:Requested authenticator <certbot.cli._Default object at 0x7f31b42690d0> and installer <certbot.cli._Default object at 0x7f31b42690d0>
2018-02-22 14:27:51,036:DEBUG:certbot.cli:Default Detector is Namespace(account=<certbot.cli._Default object at 0x7f31b42afdd0>, agree_dev_preview=None, allow_subset_of_names=<certbot.cli._Default object at 0x7f31b42af210>, apache=<certbot.cli._Default object at 0x7f31b4269e50>, apache_challenge_location=<certbot.cli._Default object at 0x7f31b42714d0>, apache_ctl=<certbot.cli._Default object at 0x7f31b42718d0>, apache_dismod=<certbot.cli._Default object at 0x7f31b426eed0>, apache_enmod=<certbot.cli._Default object at 0x7f31b426edd0>, apache_handle_modules=<certbot.cli._Default object at 0x7f31b4271650>, apache_handle_sites=<certbot.cli._Default object at 0x7f31b42717d0>, apache_init_script=<certbot.cli._Default object at 0x7f31b4271a10>, apache_le_vhost_ext=<certbot.cli._Default object at 0x7f31b4271050>, apache_logs_root=<certbot.cli._Default object at 0x7f31b4271390>, apache_server_root=<certbot.cli._Default object at 0x7f31b4271190>, apache_vhost_root=<certbot.cli._Default object at 0x7f31b4271290>, authenticator=<certbot.cli._Default object at 0x7f31b42690d0>, break_my_certs=<certbot.cli._Default object at 0x7f31b424ac10>, cert_path=<certbot.cli._Default object at 0x7f31b429f790>, certname=<certbot.cli._Default object at 0x7f31b428f2d0>, chain_path=<certbot.cli._Default object at 0x7f31b4269ad0>, checkpoints=<certbot.cli._Default object at 0x7f31b42af890>, config_dir=<certbot.cli._Default object at 0x7f31b42698d0>, config_file=None, configurator=<certbot.cli._Default object at 0x7f31b42690d0>, csr=<certbot.cli._Default object at 0x7f31b42aff90>, debug=<certbot.cli._Default object at 0x7f31b424a510>, debug_challenges=<certbot.cli._Default object at 0x7f31b424a610>, delete_after_revoke=<certbot.cli._Default object at 0x7f31b42af310>, deploy_hook=<certbot.cli._Default object at 0x7f31b424aed0>, dialog=None, directory_hooks=<certbot.cli._Default object at 0x7f31b424aad0>, dns_cloudflare=<certbot.cli._Default object at 0x7f31b426e390>, dns_cloudxns=<certbot.cli._Default object at 0x7f31b426e490>, dns_digitalocean=<certbot.cli._Default object at 0x7f31b426e590>, dns_dnsimple=<certbot.cli._Default object at 0x7f31b426e690>, dns_dnsmadeeasy=<certbot.cli._Default object at 0x7f31b426e790>, dns_google=<certbot.cli._Default object at 0x7f31b426e890>, dns_luadns=<certbot.cli._Default object at 0x7f31b426e990>, dns_nsone=<certbot.cli._Default object at 0x7f31b426ea90>, dns_rfc2136=<certbot.cli._Default object at 0x7f31b426eb90>, dns_route53=<certbot.cli._Default object at 0x7f31b426ec90>, domains=<certbot.cli._Default object at 0x7f31b429fb90>, dry_run=<certbot.cli._Default object at 0x7f31b428f410>, duplicate=<certbot.cli._Default object at 0x7f31b42afed0>, eff_email=<certbot.cli._Default object at 0x7f31b428fe50>, email=<certbot.cli._Default object at 0x7f31b428fc10>, expand=<certbot.cli._Default object at 0x7f31b42af8d0>, force_interactive=<certbot.cli._Default object at 0x7f31b429f950>, fullchain_path=<certbot.cli._Default object at 0x7f31b429ff10>, func=<function renew at 0x7f31b44cc938>, hsts=<certbot.cli._Default object at 0x7f31b4269150>, http01_address=<certbot.cli._Default object at 0x7f31b424ab10>, http01_port=<certbot.cli._Default object at 0x7f31b424aa10>, ifaces=<certbot.cli._Default object at 0x7f31b428f610>, init=<certbot.cli._Default object at 0x7f31b428ff10>, installer=<certbot.cli._Default object at 0x7f31b42690d0>, key_path=<certbot.cli._Default object at 0x7f31b429f310>, logs_dir=<certbot.cli._Default object at 0x7f31b42694d0>, manual=<certbot.cli._Default object at 0x7f31b426e190>, manual_auth_hook=<certbot.cli._Default object at 0x7f31b426ed90>, manual_cleanup_hook=<certbot.cli._Default object at 0x7f31b4271c50>, manual_public_ip_logging_ok=<certbot.cli._Default object at 0x7f31b4271d50>, max_log_backups=<certbot.cli._Default object at 0x7f31b429f4d0>, must_staple=<certbot.cli._Default object at 0x7f31b424ae10>, nginx=<certbot.cli._Default object at 0x7f31b4269f50>, nginx_ctl=<certbot.cli._Default object at 0x7f31b4271f90>, nginx_server_root=<certbot.cli._Default object at 0x7f31b4271b10>, no_bootstrap=<certbot.cli._Default object at 0x7f31b424a210>, no_self_upgrade=<certbot.cli._Default object at 0x7f31b424a110>, no_verify_ssl=<certbot.cli._Default object at 0x7f31b424a710>, noninteractive_mode=<certbot.cli._Default object at 0x7f31b429f710>, num=<certbot.cli._Default object at 0x7f31b424a5d0>, os_packages_only=<certbot.cli._Default object at 0x7f31b42affd0>, post_hook=<certbot.cli._Default object at 0x7f31b4269a50>, pre_hook=<certbot.cli._Default object at 0x7f31b4269950>, pref_challs=<certbot.cli._Default object at 0x7f31b4269850>, prepare=<certbot.cli._Default object at 0x7f31b428fa90>, quiet=<certbot.cli._Default object at 0x7f31b424a310>, reason=<certbot.cli._Default object at 0x7f31b42afd90>, redirect=<certbot.cli._Default object at 0x7f31b424af10>, register_unsafely_without_email=<certbot.cli._Default object at 0x7f31b428f790>, reinstall=<certbot.cli._Default object at 0x7f31b42af9d0>, renew_by_default=<certbot.cli._Default object at 0x7f31b42af590>, renew_hook=<certbot.cli._Default object at 0x7f31b4269b50>, renew_with_new_domains=<certbot.cli._Default object at 0x7f31b42af3d0>, rsa_key_size=<certbot.cli._Default object at 0x7f31b424ad10>, server=<certbot.cli._Default object at 0x7f31b42692d0>, staging=<certbot.cli._Default object at 0x7f31b424a410>, standalone=<certbot.cli._Default object at 0x7f31b426e090>, standalone_supported_challenges=<certbot.cli._Default object at 0x7f31b42760d0>, staple=<certbot.cli._Default object at 0x7f31b4269550>, strict_permissions=<certbot.cli._Default object at 0x7f31b4269750>, text_mode=<certbot.cli._Default object at 0x7f31b429f290>, tls_sni_01_address=<certbot.cli._Default object at 0x7f31b424a910>, tls_sni_01_port=<certbot.cli._Default object at 0x7f31b424a810>, tos=<certbot.cli._Default object at 0x7f31b42af090>, uir=<certbot.cli._Default object at 0x7f31b4269350>, update_registration=<certbot.cli._Default object at 0x7f31b428f9d0>, user_agent=<certbot.cli._Default object at 0x7f31b424a3d0>, user_agent_comment=<certbot.cli._Default object at 0x7f31b424a1d0>, validate_hooks=<certbot.cli._Default object at 0x7f31b424acd0>, verb='renew', verbose_count=<certbot.cli._Default object at 0x7f31b429fe50>, webroot=<certbot.cli._Default object at 0x7f31b426e290>, webroot_map=<certbot.cli._Default object at 0x7f31b42762d0>, webroot_path=<certbot.cli._Default object at 0x7f31b4271e90>, work_dir=<certbot.cli._Default object at 0x7f31b42696d0>)
2018-02-22 14:27:51,042:INFO:certbot.renewal:Cert not yet due for renewal
2018-02-22 14:27:51,047:INFO:certbot.renewal:Cert not yet due for renewal
2018-02-22 14:27:51,047:DEBUG:certbot.renewal:no renewal failures
2018-03-19 08:04:31,363:DEBUG:certbot.main:certbot version: 0.21.1
2018-03-19 08:04:31,364:DEBUG:certbot.main:Arguments: ['--nginx']
2018-03-19 08:04:31,364:DEBUG:certbot.main:Discovered plugins: PluginsRegistry(PluginEntryPoint#manual,PluginEntryPoint#nginx,PluginEntryPoint#null,PluginEntryPoint#standalone,PluginEntryPoint#webroot)
2018-03-19 08:04:31,370:DEBUG:certbot.log:Root logging level set at 20
2018-03-19 08:04:31,371:INFO:certbot.log:Saving debug log to /var/log/letsencrypt/letsencrypt.log
2018-03-19 08:04:31,372:DEBUG:certbot.plugins.selection:Requested authenticator nginx and installer nginx
2018-03-19 08:04:31,482:DEBUG:certbot.log:Exiting abnormally:
Traceback (most recent call last):
  File "/usr/bin/certbot", line 11, in <module>
    load_entry_point('certbot==0.21.1', 'console_scripts', 'certbot')()
  File "/usr/lib/python3/dist-packages/certbot/main.py", line 1240, in main
    return config.func(config, plugins)
  File "/usr/lib/python3/dist-packages/certbot/main.py", line 1099, in certonly
    installer, auth = plug_sel.choose_configurator_plugins(config, plugins, "certonly")
  File "/usr/lib/python3/dist-packages/certbot/plugins/selection.py", line 192, in choose_configurator_plugins
    installer = pick_installer(config, req_inst, plugins)
  File "/usr/lib/python3/dist-packages/certbot/plugins/selection.py", line 32, in pick_installer
    config, default, plugins, question, (interfaces.IInstaller,))
  File "/usr/lib/python3/dist-packages/certbot/plugins/selection.py", line 77, in pick_plugin
    verified.prepare()
  File "/usr/lib/python3/dist-packages/certbot/plugins/disco.py", line 248, in prepare
    return [plugin_ep.prepare() for plugin_ep in six.itervalues(self._plugins)]
  File "/usr/lib/python3/dist-packages/certbot/plugins/disco.py", line 248, in <listcomp>
    return [plugin_ep.prepare() for plugin_ep in six.itervalues(self._plugins)]
  File "/usr/lib/python3/dist-packages/certbot/plugins/disco.py", line 130, in prepare
    self._initialized.prepare()
  File "/usr/lib/python3/dist-packages/certbot_nginx/configurator.py", line 131, in prepare
    self.parser = parser.NginxParser(self.conf('server-root'))
  File "/usr/lib/python3/dist-packages/certbot_nginx/parser.py", line 38, in __init__
    self.load()
  File "/usr/lib/python3/dist-packages/certbot_nginx/parser.py", line 45, in load
    self._parse_recursively(self.config_root)
  File "/usr/lib/python3/dist-packages/certbot_nginx/parser.py", line 66, in _parse_recursively
    self._parse_recursively(subentry[1])
  File "/usr/lib/python3/dist-packages/certbot_nginx/parser.py", line 56, in _parse_recursively
    trees = self._parse_files(filepath)
  File "/usr/lib/python3/dist-packages/certbot_nginx/parser.py", line 206, in _parse_files
    parsed = nginxparser.load(_file)
  File "/usr/lib/python3/dist-packages/certbot_nginx/nginxparser.py", line 123, in load
    return loads(_file.read())
  File "/usr/lib/python3.5/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe8 in position 1314: ordinal not in range(128)
2018-03-19 08:04:31,483:ERROR:certbot.log:An unexpected error occurred:
2018-03-19 08:13:59,650:DEBUG:certbot.main:certbot version: 0.21.1
2018-03-19 08:13:59,650:DEBUG:certbot.main:Arguments: ['--nginx']
2018-03-19 08:13:59,651:DEBUG:certbot.main:Discovered plugins: PluginsRegistry(PluginEntryPoint#manual,PluginEntryPoint#nginx,PluginEntryPoint#null,PluginEntryPoint#standalone,PluginEntryPoint#webroot)
2018-03-19 08:13:59,658:DEBUG:certbot.log:Root logging level set at 20
2018-03-19 08:13:59,659:INFO:certbot.log:Saving debug log to /var/log/letsencrypt/letsencrypt.log
2018-03-19 08:13:59,659:DEBUG:certbot.plugins.selection:Requested authenticator nginx and installer nginx
2018-03-19 08:13:59,774:DEBUG:certbot.log:Exiting abnormally:
Traceback (most recent call last):
  File "/usr/bin/certbot", line 11, in <module>
    load_entry_point('certbot==0.21.1', 'console_scripts', 'certbot')()
  File "/usr/lib/python3/dist-packages/certbot/main.py", line 1240, in main
    return config.func(config, plugins)
  File "/usr/lib/python3/dist-packages/certbot/main.py", line 981, in run
    installer, authenticator = plug_sel.choose_configurator_plugins(config, plugins, "run")
  File "/usr/lib/python3/dist-packages/certbot/plugins/selection.py", line 189, in choose_configurator_plugins
    authenticator = installer = pick_configurator(config, req_inst, plugins)
  File "/usr/lib/python3/dist-packages/certbot/plugins/selection.py", line 25, in pick_configurator
    (interfaces.IAuthenticator, interfaces.IInstaller))
  File "/usr/lib/python3/dist-packages/certbot/plugins/selection.py", line 77, in pick_plugin
    verified.prepare()
  File "/usr/lib/python3/dist-packages/certbot/plugins/disco.py", line 248, in prepare
    return [plugin_ep.prepare() for plugin_ep in six.itervalues(self._plugins)]
  File "/usr/lib/python3/dist-packages/certbot/plugins/disco.py", line 248, in <listcomp>
    return [plugin_ep.prepare() for plugin_ep in six.itervalues(self._plugins)]
  File "/usr/lib/python3/dist-packages/certbot/plugins/disco.py", line 130, in prepare
    self._initialized.prepare()
  File "/usr/lib/python3/dist-packages/certbot_nginx/configurator.py", line 131, in prepare
    self.parser = parser.NginxParser(self.conf('server-root'))
  File "/usr/lib/python3/dist-packages/certbot_nginx/parser.py", line 38, in __init__
    self.load()
  File "/usr/lib/python3/dist-packages/certbot_nginx/parser.py", line 45, in load
    self._parse_recursively(self.config_root)
  File "/usr/lib/python3/dist-packages/certbot_nginx/parser.py", line 66, in _parse_recursively
    self._parse_recursively(subentry[1])
  File "/usr/lib/python3/dist-packages/certbot_nginx/parser.py", line 56, in _parse_recursively
    trees = self._parse_files(filepath)
  File "/usr/lib/python3/dist-packages/certbot_nginx/parser.py", line 206, in _parse_files
    parsed = nginxparser.load(_file)
  File "/usr/lib/python3/dist-packages/certbot_nginx/nginxparser.py", line 123, in load
    return loads(_file.read())
  File "/usr/lib/python3.5/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe8 in position 1314: ordinal not in range(128)
2018-03-19 08:13:59,775:ERROR:certbot.log:An unexpected error occurred:
```

my server is `ubuntu16.04`

**Comment by hackerys** [comment]
I also encountered this problem,my server is CentOS/RHEL 7,python 2.7
`2018-03-19 10:23:51,008:DEBUG:certbot.main:certbot version: 0.22.0
2018-03-19 10:23:51,008:DEBUG:certbot.main:Arguments: ['--nginx']
2018-03-19 10:23:51,008:DEBUG:certbot.main:Discovered plugins: PluginsRegistry(PluginEntryPoint#manual,PluginEntryPoint#nginx,PluginEntryPoint#null,PluginEntryPoint#standalone,PluginEntryPoint#webroot)
2018-03-19 10:23:51,028:DEBUG:certbot.log:Root logging level set at 20
2018-03-19 10:23:51,028:INFO:certbot.log:Saving debug log to /var/log/letsencrypt/letsencrypt.log
2018-03-19 10:23:51,029:DEBUG:certbot.plugins.selection:Requested authenticator nginx and installer nginx
2018-03-19 10:23:51,193:DEBUG:certbot.plugins.selection:Single candidate plugin: * nginx
Description: Nginx Web Server plugin - Alpha
Interfaces: IAuthenticator, IInstaller, IPlugin
Entry point: nginx = certbot_nginx.configurator:NginxConfigurator
Initialized: <certbot_nginx.configurator.NginxConfigurator object at 0x2f2a050>
Prep: True
2018-03-19 10:23:51,194:DEBUG:certbot.plugins.selection:Selected authenticator <certbot_nginx.configurator.NginxConfigurator object at 0x2f2a050> and installer <certbot_nginx.configurator.NginxConfigurator object at 0x2f2a050>
2018-03-19 10:23:51,194:INFO:certbot.plugins.selection:Plugins selected: Authenticator nginx, Installer nginx
2018-03-19 10:23:51,223:DEBUG:certbot.main:Picked account: <Account(RegistrationResource(body=Registration(status=u'valid', terms_of_service_agreed=None, contact=(u'mailto:yansheng@fotoable.com',), agreement=u'https://letsencrypt.org/documents/LE-SA-v1.2-November-15-2017.pdf', key=JWKRSA(key=<ComparableRSAKey(<cryptography.hazmat.backends.openssl.rsa._RSAPublicKey object at 0x237de10>)>)), uri=u'https://acme-v01.api.letsencrypt.org/acme/reg/31449609', new_authzr_uri=u'https://acme-v01.api.letsencrypt.org/acme/new-authz', terms_of_service=u'https://letsencrypt.org/documents/LE-SA-v1.2-November-15-2017.pdf'), 66e594c6fea51a5841105bb187374327, Meta(creation_host=u'wt-gameplus', creation_dt=datetime.datetime(2018, 3, 19, 10, 20, 44, tzinfo=<UTC>)))>
2018-03-19 10:23:51,229:DEBUG:acme.client:Sending GET request to https://acme-v01.api.letsencrypt.org/directory.
2018-03-19 10:23:51,233:INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): acme-v01.api.letsencrypt.org
2018-03-19 10:23:51,508:DEBUG:requests.packages.urllib3.connectionpool:"GET /directory HTTP/1.1" 200 562
2018-03-19 10:23:51,509:DEBUG:acme.client:Received response:
HTTP 200
content-length: 562
expires: Mon, 19 Mar 2018 10:23:51 GMT
strict-transport-security: max-age=604800
server: nginx
connection: keep-alive
pragma: no-cache
cache-control: max-age=0, no-cache, no-store
date: Mon, 19 Mar 2018 10:23:51 GMT
x-frame-options: DENY
content-type: application/json
replay-nonce: a7vIGj3X3kj-ua5hfzgyGKSZe08DWqiqu7eltjzBY0Q

{
  "key-change": "https://acme-v01.api.letsencrypt.org/acme/key-change",
  "meta": {
    "terms-of-service": "https://letsencrypt.org/documents/LE-SA-v1.2-November-15-2017.pdf"
  },
  "new-authz": "https://acme-v01.api.letsencrypt.org/acme/new-authz",
  "new-cert": "https://acme-v01.api.letsencrypt.org/acme/new-cert",
  "new-reg": "https://acme-v01.api.letsencrypt.org/acme/new-reg",
  "revoke-cert": "https://acme-v01.api.letsencrypt.org/acme/revoke-cert",
  "wo7nrI3-yI4": "https://community.letsencrypt.org/t/adding-random-entries-to-the-directory/33417"
}
2018-03-19 10:23:51,515:DEBUG:certbot.util:Not suggesting name "_"
2018-03-19 10:23:51,515:DEBUG:certbot.util:_ contains an invalid character. Valid characters are A-Z, a-z, 0-9, ., and -.
2018-03-19 10:23:58,967:INFO:certbot.main:Obtaining a new certificate
2018-03-19 10:23:59,045:DEBUG:certbot.crypto_util:Generating key (2048 bits): /etc/letsencrypt/keys/0002_key-certbot.pem
2018-03-19 10:23:59,047:DEBUG:certbot.crypto_util:Creating CSR: /etc/letsencrypt/csr/0002_csr-certbot.pem
2018-03-19 10:23:59,047:DEBUG:acme.client:Requesting fresh nonce
2018-03-19 10:23:59,047:DEBUG:acme.client:Sending HEAD request to https://acme-v01.api.letsencrypt.org/acme/new-authz.
2018-03-19 10:23:59,287:DEBUG:requests.packages.urllib3.connectionpool:"HEAD /acme/new-authz HTTP/1.1" 405 0
2018-03-19 10:23:59,288:DEBUG:acme.client:Received response:
HTTP 405
content-length: 91
pragma: no-cache
expires: Mon, 19 Mar 2018 10:23:59 GMT
server: nginx
connection: keep-alive
allow: POST
cache-control: max-age=0, no-cache, no-store
date: Mon, 19 Mar 2018 10:23:59 GMT
content-type: application/problem+json
replay-nonce: n247cQNmViYazdX6TKIhAAOE65nMoszghkNynP59ug8


2018-03-19 10:23:59,288:DEBUG:acme.client:Storing nonce: n247cQNmViYazdX6TKIhAAOE65nMoszghkNynP59ug8
2018-03-19 10:23:59,288:DEBUG:acme.client:JWS payload:
{
  "identifier": {
    "type": "dns", 
    "value": "wt.gameplus.io"
  }, 
  "resource": "new-authz"
}
2018-03-19 10:23:59,291:DEBUG:acme.client:Sending POST request to https://acme-v01.api.letsencrypt.org/acme/new-authz:
{
  "protected": "eyJub25jZSI6ICJuMjQ3Y1FObVZpWWF6ZFg2VEtJaEFBT0U2NW5Nb3N6Z2hrTnluUDU5dWc4IiwgImFsZyI6ICJSUzI1NiIsICJqd2siOiB7ImUiOiAiQVFBQiIsICJrdHkiOiAiUlNBIiwgIm4iOiAieVpmWnAwdkd0NFFjbGRnMExuT2xmckE4bHBSZ1VJc2t5anVhUkRqVXpQM3RTSXRTS2VSUUNhekhVdkdCUXBfdmdrNE9YRm9kODJ2UkRrMlA0amtudmtUZHF6Q1gyN096SnQ4MlI3dFU1eW5YUW1aNzJwcThwUERLN0llZVNucXVidGhrS2J4RU44V1VRdWFxRGpQSVpFY3dnQktaTDh0aUYtTEdRaHVfZXVwQ2tzS2tlOWh4cU9mdFJMOWZQNFZXQmlVUlU0NTUwclltSzlpM3RvX0l3U1I3Y0RNX0RtRTg4MXF6T3pwdDVlUTRlVVZicURGb0Etb2t1MG9Xb3ZPcXJrcHBfMXVqQzFVS3BnTWl2ZFFIVnVHbWhMUVhyQ2hqNVNrblZ1Z2pPQ1ZUVnVVcGJ2ekZhd3d1QVlxM1pmenRLUWYxTzlHV3F6R0JoMFlIMGpBSGRRIn19", 
  "payload": "ewogICJpZGVudGlmaWVyIjogewogICAgInR5cGUiOiAiZG5zIiwgCiAgICAidmFsdWUiOiAid3QuZ2FtZXBsdXMuaW8iCiAgfSwgCiAgInJlc291cmNlIjogIm5ldy1hdXRoeiIKfQ", 
  "signature": "oWcRmpqtzyTKDtN35RGFoKm9gZlIhABLz0P5KxRXISCKJsSKunSEcOTu8lC8FjnZBYjko3xZN4wT5zneRmS2__brFOk2sj63M_WTWNaaBeP9A3pjWnqutwW-DlZNs8qAZTl4Q7BW9BAzn84QrQlGzAu4HsR4FPPmemyi_hnhU9LYPd9z5Mnwdn5HhDp6nC20F5xrTZjmwO8Se6aSa8r-O8_Hnx3yllKyGnqWUnr18TG1vkb00qbznVL-LauDqyvFbF5VRT15DksAMzd_U4JMxQuxgmpVsgNNPLVmteXX1r9xpv_uysV4NCb2MP5eyaD8zgzp_0ENsWJT5lUxoHJedQ"
}
2018-03-19 10:23:59,634:DEBUG:requests.packages.urllib3.connectionpool:"POST /acme/new-authz HTTP/1.1" 201 729
2018-03-19 10:23:59,634:DEBUG:acme.client:Received response:
HTTP 201
content-length: 729
expires: Mon, 19 Mar 2018 10:23:59 GMT
cache-control: max-age=0, no-cache, no-store
strict-transport-security: max-age=604800
server: nginx
connection: keep-alive
link: <https://acme-v01.api.letsencrypt.org/acme/new-cert>;rel="next"
location: https://acme-v01.api.letsencrypt.org/acme/authz/NfZEck_Gu6MAP4q4ti4S_3c77ndu4e62daMinr-fwgQ
pragma: no-cache
boulder-requester: 31449609
date: Mon, 19 Mar 2018 10:23:59 GMT
x-frame-options: DENY
content-type: application/json
replay-nonce: FZLgk_Gr9g7fsldiYFaZE01IyNo1w4MT1ItqZtKyRMs

{
  "identifier": {
    "type": "dns",
    "value": "wt.gameplus.io"
  },
  "status": "pending",
  "expires": "2018-03-26T10:23:59.456530831Z",
  "challenges": [
    {
      "type": "http-01",
      "status": "pending",
      "uri": "https://acme-v01.api.letsencrypt.org/acme/challenge/NfZEck_Gu6MAP4q4ti4S_3c77ndu4e62daMinr-fwgQ/3877005638",
      "token": "1Q5WXt-xfZHGwuje7wzP7TJMHJpkw8YXBhEr1vtWnvg"
    },
    {
      "type": "dns-01",
      "status": "pending",
      "uri": "https://acme-v01.api.letsencrypt.org/acme/challenge/NfZEck_Gu6MAP4q4ti4S_3c77ndu4e62daMinr-fwgQ/3877005639",
      "token": "Ob8mYTw-94dkDAyi0k_KizOGFUkLcSgrnr8Q71YKRHo"
    }
  ],
  "combinations": [
    [
      0
    ],
    [
      1
    ]
  ]
}
2018-03-19 10:23:59,634:DEBUG:acme.client:Storing nonce: FZLgk_Gr9g7fsldiYFaZE01IyNo1w4MT1ItqZtKyRMs
2018-03-19 10:23:59,635:INFO:certbot.auth_handler:Performing the following challenges:
2018-03-19 10:23:59,635:INFO:certbot.auth_handler:http-01 challenge for wt.gameplus.io
2018-03-19 10:23:59,663:DEBUG:certbot.reverter:Creating backup of /etc/nginx/nginx.conf
2018-03-19 10:23:59,663:DEBUG:certbot.reverter:Creating backup of /etc/nginx/enable-php.conf
2018-03-19 10:23:59,663:DEBUG:certbot.reverter:Creating backup of /etc/nginx/mime.types
2018-03-19 10:23:59,667:DEBUG:certbot.error_handler:Encountered exception:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/certbot/auth_handler.py", line 127, in _solve_challenges
    resp = self.auth.perform(all_achalls)
  File "/usr/lib/python2.7/site-packages/certbot_nginx/configurator.py", line 969, in perform
    http_response = http_doer.perform()
  File "/usr/lib/python2.7/site-packages/certbot_nginx/http_01.py", line 61, in perform
    self.configurator.save("HTTP Challenge", True)
  File "/usr/lib/python2.7/site-packages/certbot_nginx/configurator.py", line 899, in save
    self.parser.filedump(ext='')
  File "/usr/lib/python2.7/site-packages/certbot_nginx/parser.py", line 242, in filedump
    out = nginxparser.dumps(tree)
  File "/usr/lib/python2.7/site-packages/certbot_nginx/nginxparser.py", line 134, in dumps
    return str(RawNginxDumper(blocks.spaced))
  File "/usr/lib/python2.7/site-packages/certbot_nginx/nginxparser.py", line 98, in __str__
    return ''.join(self)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe6 in position 51: ordinal not in range(128)

2018-03-19 10:23:59,667:DEBUG:certbot.error_handler:Calling registered functions
2018-03-19 10:23:59,667:INFO:certbot.auth_handler:Cleaning up challenges
2018-03-19 10:24:00,806:DEBUG:certbot.log:Exiting abnormally:
Traceback (most recent call last):
  File "/bin/certbot", line 9, in <module>
    load_entry_point('certbot==0.22.0', 'console_scripts', 'certbot')()
  File "/usr/lib/python2.7/site-packages/certbot/main.py", line 1266, in main
    return config.func(config, plugins)
  File "/usr/lib/python2.7/site-packages/certbot/main.py", line 1031, in run
    certname, lineage)
  File "/usr/lib/python2.7/site-packages/certbot/main.py", line 118, in _get_and_save_cert
    lineage = le_client.obtain_and_enroll_certificate(domains, certname)
  File "/usr/lib/python2.7/site-packages/certbot/client.py", line 350, in obtain_and_enroll_certificate
    cert, chain, key, _ = self.obtain_certificate(domains)
  File "/usr/lib/python2.7/site-packages/certbot/client.py", line 294, in obtain_certificate
    orderr = self._get_order_and_authorizations(csr.data, self.config.allow_subset_of_names)
  File "/usr/lib/python2.7/site-packages/certbot/client.py", line 330, in _get_order_and_authorizations
    authzr = self.auth_handler.handle_authorizations(orderr, best_effort)
  File "/usr/lib/python2.7/site-packages/certbot/auth_handler.py", line 75, in handle_authorizations
    resp = self._solve_challenges()
  File "/usr/lib/python2.7/site-packages/certbot/auth_handler.py", line 127, in _solve_challenges
    resp = self.auth.perform(all_achalls)
  File "/usr/lib/python2.7/site-packages/certbot_nginx/configurator.py", line 969, in perform
    http_response = http_doer.perform()
  File "/usr/lib/python2.7/site-packages/certbot_nginx/http_01.py", line 61, in perform
    self.configurator.save("HTTP Challenge", True)
  File "/usr/lib/python2.7/site-packages/certbot_nginx/configurator.py", line 899, in save
    self.parser.filedump(ext='')
  File "/usr/lib/python2.7/site-packages/certbot_nginx/parser.py", line 242, in filedump
    out = nginxparser.dumps(tree)
  File "/usr/lib/python2.7/site-packages/certbot_nginx/nginxparser.py", line 134, in dumps
    return str(RawNginxDumper(blocks.spaced))
  File "/usr/lib/python2.7/site-packages/certbot_nginx/nginxparser.py", line 98, in __str__
    return ''.join(self)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe6 in position 51: ordinal not in range(128)
2018-03-19 10:24:00,808:ERROR:certbot.log:An unexpected error occurred:`

**Comment by schoen** [comment]
Thanks for reporting this. This is a duplicate of issue https://github.com/certbot/certbot/issues/5646. You can watch that issue to see our progress on fixing this.

A workaround is usually to remove non-ASCII characters in comments (or elsewhere) in nginx configuration files. However, this behavior by Certbot is definitely a bug because non-ASCII characters should certainly be permitted in configuration files—they're allowed by nginx.

**Comment by EgorZotov** [comment]
@hackerys @JZLeung  
You can use tools such as grep for looking proplematic places.
This command helped me to workaround this issue :
`grep -r -P -n '[^\x00-\x7f]' /etc/nginx`

**Comment by bigape** [comment]
Thanks Egor. Was able to find my issue. For some reason it was an apostrophe in a sites-enabled comment "Don▒t cache uris containing the following segments."
, showed up as a domino in the grep output. Apparently I copied and pasted from a source that used an RTL or East Asian apostrophe or something weird. Another file that was filled with non-UTF characters was a Nench Bench server test file - also in sites-enabled. I deleted that and removed the non-UTF apostrophe that showed up fine in nano, Notepad++, and WinSCP text editor.  Certbot was able to generate properly afterwards. For a fix in certbot, should attempt to load more python modules before a program fail termination.
